vita3k/https/src/https.cpp

```diff
@@ -236,285 +63,94 @@ std::string get_web_regex_result(const std::string &url, const std::regex &regex
-                    remaining_time = static_cast<uint64_t>((remaining_bytes / downloaded_bytes) * elapsed_time_ms) / 1000;
-                }
+        while (callback_result->pause) {
+            // Check if the download has been aborted
+            if (!callback_result->download) {
+                return 1; // Returning anything thats not 0 aborts the request
             }
-        } else {
-            // Sleep for 100ms to not consume CPU
-            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        };
 
-            // Reset initial downloaded file size and start time to calculate remaining time correctly when resume download
-            initial_downloaded_file_size = downloaded_file_size;
-            start_time = get_current_time_ms();
+        // Also check here just in case pause was set to false first
+        if (!callback_result->download) {
+            return 1; // Returning anything thats not 0 aborts the request
         }
+        return 0;
+    };
 
-        // Call progress callback function to update progress info and state
-        if (progress_callback)
-            progress_state = progress_callback(progress_percent, remaining_time);
+    auto start_time = get_current_time_ms();
+    auto callbackData = CallbackData(start_time, progress_callback);
+
+    curl_easy_setopt(curl_download, CURLOPT_URL, url.c_str());
+    curl_easy_setopt(curl_download, CURLOPT_WRITEFUNCTION, write_data);
+    curl_easy_setopt(curl_download, CURLOPT_NOPROGRESS, false); // Enable progress function
+    curl_easy_setopt(curl_download, CURLOPT_XFERINFODATA, &callbackData);
+    curl_easy_setopt(curl_download, CURLOPT_XFERINFOFUNCTION, curl_callback);
+    curl_easy_setopt(curl_download, CURLOPT_FOLLOWLOCATION, true); // Follow redirects
+
+    auto fp = fopen(output_file_path.c_str(), "wb");
+    if (!fp) {
+        LOG_CRITICAL("Could not fopen file {}", output_file_path);
+        return false;
     }
 
-    // Close the output file
-    outfile.close();
-
-    // Close SSL and socket connection
-    close_ssl(https.ssl);
-    close_socket(https.sockfd);
+    curl_easy_setopt(curl_download, CURLOPT_WRITEDATA, fp);
+    int res = curl_easy_perform(curl_download);
 
-    // Reset progress state to 0
-    if (progress_callback)
-        progress_state = progress_callback(0, 0);
-
-    // Check if download file size is same of file size
-    if (downloaded_file_size < (file_size * 0.99)) {
-        if (progress_state.download)
-            LOG_ERROR("Downloaded size is not equal to file size, downloaded size: {}/{}", downloaded_file_size, file_size);
-        else
-            LOG_WARN("Canceled by user, dowloaded size: {}/{}", downloaded_file_size, file_size);
-        return false;
-    }
+    fclose(fp);
+    curl_easy_cleanup(curl_download);
+    progress_callback(0, 0);
 
-    // Check if the downloaded file is corrupted
-    const auto downloaded_file_md5 = calculate_md5_file(output_file_path);
```

Comment:
Don't remove my code of check and verify md5
I have take time for coding it...
I see in curl you can get header, using it